sycl/include/CL/sycl/group_algorithm.hpp

```diff
@@ -97,6 +101,37 @@ template <typename T, typename BinaryOperation> struct is_native_op {
       is_contained<BinaryOperation, native_op_list<void>>::value;
 };
 
+// ---- is_complex
+template <typename T>
+struct is_complex
+    : std::integral_constant<
+          bool, std::is_same<T, std::complex<float>>::value ||
+                    std::is_same<T, std::complex<double>>::value ||
+                    std::is_same<T, std::complex<long double>>::value> {};
+
+// ---- is_arithmetic_or_complex
+template <typename T>
+using is_arithmetic_or_complex =
+    std::integral_constant<bool, sycl::detail::is_complex<T>::value ||
+                                     sycl::detail::is_arithmetic<T>::value>;
+
+// ---- identity_for_ga_op
+//   the group algorithms support std::complex, limited to sycl::plus operation
+//   get the correct identity for group algorithm operation.
+template <typename T, class BinaryOperation>
+constexpr detail::enable_if_t<
+    (is_complex<T>::value &&
+     std::is_same<BinaryOperation, sycl::plus<T>>::value),
+    T>
+identity_for_ga_op() {
+  return {0, 0};
+}
```

Comment:
We should consider whether it makes sense to extend `sycl::known_identity_v` to `std::complex` types.  I'd like to hear from @v-klochkov -- how hard would it be to do this?

Reply:
@Pennycook the reason I didn't extend known_identity was that I didn't want to accidentally introduce complex support to a bunch of places where it might not be intended (all the callers of known_identity), plus this particular identity of {0,0} only works because the only operation we are worried about is plus.  But that wouldn't be correct for the more general known_identity.

That said, I guess it's worth asking.